File: Application/app/src/main/java/be/eaict/blackboardsnapshotapp/Objects/Repository.java
package be.eaict.blackboardsnapshotapp.Objects;

public class Repository implements IRepository   {

    private static IRepository repo = null;
    protected DataFile photos;

    public static IRepository getInstance(){
        if(repo == null){
            repo = new Repository();
        }
        return repo;
    }

    public void setPhotos(DataFile data){
        this.photos = data;
    }

    public DataFile getPhotos(){

        return photos;
    }
}

File: Application/app/src/main/java/be/eaict/blackboardsnapshotapp/Objects/Repository.kt
package be.eaict.blackboardsnapshotapp.Objects

import android.provider.ContactsContract
import java.util.ArrayList

class Repository : IRepository {
    private lateinit var repo: IRepository
    val api = API()
    private lateinit var photos : DataFile

    fun getInstance(): IRepository {
        if (repo == null) {
            repo = Repository()
        }
        return repo
    }

    override fun callAPI(){
        photos = api.callAPI()
    }

    override fun getPhotos(): DataFile {

        return photos
    }
}
